src/graccreq/oim/nsfscience.py

```diff
@@ -0,0 +1,50 @@
+"""
+Add NSF Field of Science to GRACC records
+
+"""
+import csv
+import urllib2
+
+
+
+class NSFScience(object):
+    """
+    Add the OIM_NSFFieldOfScience attribute to jobs
+    """
+
+    def __init__(self, url="file:///etc/nsffieldofscience.csv"):
+        """
+
+        """
+        self.mapping_csv = url
+        self._parseFields(self.mapping_csv)
+
+
+    def _parseFields(self, url):
+        raw_csv = urllib2.urlopen(url)
+        parsed_csv = csv.reader(raw_csv, delimiter=',')
+
+        self.mapping_dict = {}
+        for row in parsed_csv:
+            self.mapping_dict[row[0]] = row[1]
+
+
+    def parseDoc(self, record):
+        """
+        Parse the current record.  Look for the attribute OIM_FieldOfScience and map
+        to the OIM_NSFFieldOfScience
+
+        :returns: Minimal dict with attributes to add
+        """
+        if 'OIM_FieldOfScience' in record:
```

Comment:
To avoid case issues, should we `.lower()` this and lower the first column of the csv?